src/signal_processing_algorithms/difference.py

```diff
@@ -0,0 +1,18 @@
+"""Calculate difference matrices."""
+import numpy as np
+
+
+def pairwise_difference(x: np.ndarray, y: np.ndarray = None) -> np.ndarray:
```

Comment:
These are pairwise differences, not just one difference, right?  Or perhaps a distance matrix.